lib/winapi/sync.c

```diff
@@ -40,6 +40,106 @@ VOID LeaveCriticalSection (LPCRITICAL_SECTION lpCriticalSection)
     RtlLeaveCriticalSection(lpCriticalSection);
 }
 
+void AcquireSRWLockExclusive (PSRWLOCK SRWLock)
+{
+    // Grab the global lock (second-highest bit)
+    uintptr_t unlocked_val = SRWLock->Ptr & 0xBFFFFFFF;
+
+    while (!__sync_bool_compare_and_swap(&SRWLock->Ptr, unlocked_val, unlocked_val | 0x40000000)) {
```

Comment:
nit: You actually do use `0x40000000` here instead of `~0xBFFFFFFF` - so it's harder to see the relation.